README.md
# Flow Theme

## General Information

*Flow* is a responsive theme for all OXID eShop editions.

**Warning!** Flow theme is depreciated. See also [Wave Theme](https://github.com/OXID-eSales/wave-theme) and [Graph QL Api](https://github.com/OXID-eSales/graphql-storefront-module) for alternatives. 

## Installation

### Step 1: Select the appropriate branch

* Branch ``b-3.x`` contains *Flow* for OXID eShop compilation 6.3 and 6.4
* Branch ``master`` contains *Flow* for OXID eShop master branch's latest commits

### Step 2: Get the source code

Check out the appropriate branch for your OXID eShop version into the shop's ``Application/views`` directory:

```
cd DOCUMENT_ROOT/Application/views/
git clone -b BRANCH_NAME https://github.com/OXID-eSales/flow_theme.git flow
```

You should now have a new folder ``DOCUMENT_ROOT/Application/views/flow``.

### Step 3: Prepare the ``out`` folder

The ``out`` folder can be prepared in two possible ways. The first option is to use symbolic links, the second is to copy. Using a symbolic link is strongly recommended, as otherwise you would have do the copying everytime *Flow* is updated.

#### Option 1: Symbolic Linking

Link the theme's ``out`` directory to the ``out`` directory of the shop:

``ln -s ../Application/views/flow/out/flow ../../out/flow``

#### Option 2: Copying

Copy theme's ``out`` directory to shop:

``cp -r flow/out/flow ../../out/flow``

You now should be able to do ``cd ../../out/flow``.

### Step 4: Activate theme 

Log into the admin panel, go to *Extensions → Themes → Flow* and press the *Activate* button. Clean the cache and off you go, my friend!

## Development

All *Flow* theme related CSS/Javascript files can be found in theme's ``build`` directory. To regenerate the theme's assets, the ``grunt`` tasks should be used. Please install ``grunt`` and run ``grunt's`` default task to regenerate all minimized ``css`` and ``js`` files:

To get the development files you need to [clone the repository](#step-2-get-the-source-code).

1. To use ``grunt``, ``npm`` is required. Check ``nodejs`` website for installation
instructions (https://nodejs.org/en/download/package-manager/). Example of
Installation on ubuntu system:

	```
	curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
	sudo apt-get install -y nodejs
	```

2. Install ``grunt`` globally and update npm. Example:

	```
    sudo npm install -g npm grunt-cli 
    ```

3. Go to "flow" theme's directory and install all related ``grunt`` plugins:

	```
    cd DOCUMENT_ROOT/Application/views/flow/
    npm install
    ```

4. Now its possible to regenerate "flow" theme assets by running ``grunt`` default
task while being in "flow" directory:

	```
	grunt
	```

## Contributing

If you are interested in contributing of some changes, Please read [CONTRIBUTING.md](CONTRIBUTING.md) file for more information.

## Distribution notice
The Flow Theme will not be shipped with the next major release of OXID eShop (>6) as it is using older components!
Please consider to use the [Wave Theme](https://github.com/OXID-eSales/wave-theme).

## Issues

Please forward all issues to https://bugs.oxid-esales.com/ for the project **Flow Theme**.


CHANGELOG.md
# Change Log for OXID eSales Flow Theme

All notable changes to this project will be documented in this file.
The format is based on [Keep a Changelog](http://keepachangelog.com/)
and this project adheres to [Semantic Versioning](http://semver.org/).

## [5.0.0] - Unreleased

### Changed
- License update

### Removed
- RSS functionality

## [4.0.0] - 2021-07-06

### Changed
- BC Break: block in ``tpl/email/plain/ordershopped.tpl`` renamed from ``email_html_ordershipped_shipmenttrackingurl`` to ``email_plain_ordershipped_shipmenttrackingurl`` [PR-153](https://github.com/OXID-eSales/flow_theme/pull/153)
- Decoded ``oxconfig`` insert queries in ``setup.sql``
- Move translations to ``translations/`` directory
- In admin settings 'Email logo' to display group 'Emails' [Wave PR-97](https://github.com/OXID-eSales/wave-theme/pull/97)

### Fixed
- Improved readability of setup.sql and theme.php [Wave PR-97](https://github.com/OXID-eSales/wave-theme/pull/97)

### Removed
- Google+ related settings together with frontend and email template blocks [PR-155](https://github.com/OXID-eSales/flow_theme/pull/155)
- Remove credit card related blocks in payment method selection
- Remove Betanote
- Suggest (Recommend Product) feature:
  - Templates:
    - `email/html/suggest.tpl`
    - `email/plain/suggest.tpl`
    - `form/suggest.tpl`
    - `page/info/suggest.tpl`
  - Link in: `page/details/inc/productmania.tpl`
  - Language Constant: `DD_SUGGEST_HEADING`
- News feature
- `setup.sql` file support and file itself
- Remove `ajax_econda_recommendations.tpl` use econda module instead

## [3.8.1] - 2022-07-18

### Fixed
- CONTRIBUTING link

## [3.8.0] - 2021-11-26

### Fixed
- Template content is not updated with dynamic caching enabled [#0005354](https://bugs.oxid-esales.com/view.php?id=5354)

### Removed
- Unused templates

## [3.7.2] - 2021-07-20

### Fixed
- Improve jQuery security

## [3.7.1] - 2021-07-06

### Fixed
- Add missing actions.tpl for Actions widget [Wave PR-121](https://github.com/OXID-eSales/wave-theme/pull/121)

## [3.7.0] - 2021-03-12

### Added
- New blocks:
  - ``captcha_form`` in ``tpl/form/forgotpwd_email.tpl`` [PR-163](https://github.com/OXID-eSales/flow_theme/pull/163)
  - ``captcha_form`` in ``tpl/form/newsletter.tpl`` [PR-163](https://github.com/OXID-eSales/flow_theme/pull/163)
  - ``salutation_options`` in ``tpl/form/fieldset/salutation.tpl`` [PR-164](https://github.com/OXID-eSales/flow_theme/pull/164)

## [3.6.0] - 2020-11-12

### Added
- New blocks:
  - ``user_account_password`` in ``tpl/form/forgotpwd_change_pwd.tpl`` [PR-154](https://github.com/OXID-eSales/flow_theme/pull/154)
  - ``user_account_password`` in ``tpl/form/user_password.tpl`` [PR-154](https://github.com/OXID-eSales/flow_theme/pull/154)

### Fixed
- Variantselector doesn't work in List View listitem_line in wave and flow [#0007163](https://bugs.oxid-esales.com/view.php?id=7163) [PR-162](https://github.com/OXID-eSales/flow_theme/pull/162)

## [3.5.1] - 2020-08-31

### Changed
- Update minified JavaScript and CSS

## [3.5.0] - 2020-07-08

### Added
- Added the checkbox to enable/disable pricealarm tab in product id [Wave PR-71](https://github.com/OXID-eSales/wave-theme/pull/71)
- Add theme option to make category menu in header fixed or not

### Fixed
- Use schema.org specification for breadcrumbs [Wave PR-83](https://github.com/OXID-eSales/wave-theme/pull/83)
- Avoid empty row before subcategories list [Wave PR-98](https://github.com/OXID-eSales/wave-theme/pull/98)
- Fix align of text in basket delivery tax row [Wave PR-100](https://github.com/OXID-eSales/wave-theme/pull/100)
- Fix wrapping in order/shipping email on small viewport [PR-158](https://github.com/OXID-eSales/flow_theme/pull/158)

### Renamed
- Changed price alert to wished price

## [3.4.1] - 2020-03-16

### Fixed
- Translate hard coded string 'Mehr' [PR-156](https://github.com/OXID-eSales/flow_theme/pull/156) [#0006521](https://bugs.oxid-esales.com/view.php?id=6521)

## [3.4.0] - 2019-10-15

### Removed
- Remove outdated econda keys from en translations [Wave PR-52](https://github.com/OXID-eSales/wave-theme/pull/52)

### Fixed
- Use sStartPageListDisplayType setting in Top seller start page block [Wave PR-57](https://github.com/OXID-eSales/wave-theme/pull/57)
- Fix compare list remove link on details page [Wave PR-59](https://github.com/OXID-eSales/wave-theme/pull/59)
- Fix first error field message disappearing right after validation [Wave PR-45](https://github.com/OXID-eSales/wave-theme/pull/45)
- Ensure that variables are assigned for widget product list and listitems.
- Add missing help-block elements for several billing/shipping address edit form fields [Wave PR-64](https://github.com/OXID-eSales/wave-theme/pull/64)
- Display netto prices in netto mode in quantity block [#0006811](https://bugs.oxid-esales.com/view.php?id=6811) [PR-151](https://github.com/OXID-eSales/flow_theme/pull/151)

### Changed
- Changed block ``user_account_newsletter`` in form/fieldset/user_account.tpl from inner form-group to outer form-group
- Don't output the legal footer if oxstdfooter isn't there [Wave PR-58](https://github.com/OXID-eSales/wave-theme/pull/58)

## [3.3.0] - 2019-07-12

### Added
- New blocks:
  - `widget/footer/services.tpl`
    - footer_services_cart
  - `form/fieldset/user_shipping.tpl`  
    - form_user_shipping_address_actions
    - form_user_shipping_address_edit_action
    - form_user_shipping_address_delete_action
- Add stoken parameter to remove voucher link

### Fixed
- Fix "sBackgroundPosVertical" config variable constraints [Wave PR-33](https://github.com/OXID-eSales/wave-theme/pull/33)
- Do not use deprecated getSession of Base class
- Fix HTML syntax error in oxaddress__oxfname req [Wave PR-41](https://github.com/OXID-eSales/wave-theme/pull/41)
- Fix wrong icon in offline page [WAVE PR-38](https://github.com/OXID-eSales/wave-theme/pull/38)

## [3.2.0] - 2019-04-11

### Added
- New blocks:
  - `page/search/search.tpl`

theme.php
<?php

/**
 * This file is part of OXID eSales Flow theme.
 *
 * OXID eSales Flow theme is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * OXID eSales Flow theme is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with OXID eSales Flow theme.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @link      http://www.oxid-esales.com
 * @copyright (C) OXID eSales AG 2003-2016
 */

/**
 * Theme Information
 */
$aTheme = array(
    'id'          => 'flow',
    'title'       => 'Flow',
    'description' => 'Flow is OXID`s official responsive theme based on the CSS framework Bootstrap 3.',
    'thumbnail'   => 'theme.jpg',
    'version'     => '4.0.0',
    'author'      => '<a href="http://www.oxid-esales.com" title="OXID eSales AG">OXID eSales AG</a>',
    'settings'    => array(
        //Group display:
        array(
            'group' => 'display',
            'name'  => 'blFullwidthLayout',
            'type'  => 'bool',
            'value' => 0,
            'position' => 1000
        ),
        array(
            'group' => 'display',
            'name'  => 'bl_showManufacturerSlider',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1005
        ),
        array(
            'group'       => 'display',
            'name'        => 'sDefaultListDisplayType',
            'type'        => 'select',
            'value'       => 'infogrid',
            'constraints' => 'infogrid|line|grid',
            'position' => 1010
        ),
        array(
            'group' => 'display',
            'name'  => 'blShowListDisplayType',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1015
        ),
        array(
            'group'       => 'display',
            'name'        => 'sStartPageListDisplayType',
            'type'        => 'select',
            'value'       => 'grid',
            'constraints' => 'infogrid|line|grid',
            'position' => 1020
        ),
        array(
            'group' => 'display',
            'name'  => 'aNrofCatArticles',
            'type'  => 'arr',
            'value' => array("10", "20", "50", "100"),
            'position' => 1025
        ),
        array(
            'group' => 'display',
            'name'  => 'aNrofCatArticlesInGrid',
            'type'  => 'arr',
            'value' => array("12", "16", "24", "32"),
            'position' => 1030
        ),
        array(
            'group'       => 'display',
            'name'        => 'iNewBasketItemMessage',
            'type'        => 'select',
            'value'       => 'str',
            'constraints' => '0|1|2|3',
            'position' => 1035
        ),
        array(
            'group' => 'display',
            'name'  => 'blShowFinalStep',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1040
        ),
        array(
            'group' => 'display',
            'name'  => 'blShowBirthdayFields',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1045
        ),

        //Group logo:
        array(
            'group' => 'logo',
            'name'  => 'sLogoFile',
            'type'  => 'str',
            'value' => 'logo_oxid.png',
            'position' => 1100
        ),
        array(
            'group' => 'logo',
            'name'  => 'sLogoWidth',
            'type'  => 'str',
            'value' => '200',
            'position' => 1105
        ),
        array(
            'group' => 'logo',
            'name'  => 'sLogoHeight',
            'type'  => 'str',
            'value' => '38',
            'position' => 1110
        ),

        //Group favicon:
        array(
            'group' => 'favicons',
            'name'  => 'sFaviconFile',
            'type'  => 'str',
            'value' => 'favicon.ico',
            'position' => 1200
        ),
        array(
            'group' => 'favicons',
            'name'  => 'sFavicon16File',
            'type'  => 'str',
            'value' => 'favicon_16x16.png',
            'position' => 1205
        ),
        array(
            'group' => 'favicons',
            'name'  => 'sFavicon32File',
            'type'  => 'str',
            'value' => 'favicon_32x32.png',
            'position' => 1210
        ),
        array(
            'group' => 'favicons',
            'name'  => 'sFavicon48File',
            'type'  => 'str',
            'value' => 'favicon_48x48.png',
            'position' => 1215
        ),
        array(
            'group' => 'favicons',
            'name'  => 'sFavicon64File',
            'type'  => 'str',
            'value' => 'favicon_64x64.png',
            'position' => 1220
        ),
        array(
            'group' => 'favicons',
            'name'  => 'sFavicon128File',
            'type'  => 'str',
            'value' => 'favicon_128x128.png',
            'position' => 1225
        ),
        array(
            'group' => 'favicons',
            'name'  => 'sFavicon512File',
            'type'  => 'str',
            'value' => 'favicon_512x512.png',
            'position' => 1230
        ),
        array(
            'group' => 'favicons',
            'name'  => 'sFaviconMSTileColor',
            'type'  => 'str',
            'value' => '#D83434',
            'position' => 1235
        ),

        //Group header:
        array(
            'group' => 'header',
            'name'  => 'stickyHeader',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1300
        ),

        //Group images:
        array(
            'group' => 'images',
            'name'  => 'sIconsize',
            'type'  => 'str',
            'value' => '87*87',
            'position' => 1400
        ),
        array(
            'group' => 'images',
            'name'  => 'sThumbnailsize',
            'type'  => 'str',
            'value' => '390*245',
            'position' => 1405
        ),
        array(
            'group' => 'images',
            'name'  => 'aDetailImageSizes',
            'type'  => 'aarr',
            'value' => array(
                'oxpic1'  => '540*340',
                'oxpic2'  => '540*340',
                'oxpic3'  => '540*340',
                'oxpic4'  => '540*340',
                'oxpic5'  => '540*340',
                'oxpic6'  => '540*340',
                'oxpic7'  => '540*340',
                'oxpic8'  => '540*340',
                'oxpic9'  => '540*340',
                'oxpic10' => '540*340',
                'oxpic11' => '540*340',
                'oxpic12' => '540*340',
            ),
            'position' => 1410
        ),
        array(
            'group' => 'images',
            'name'  => 'sZoomImageSize',
            'type'  => 'str',
            'value' => '665*665',
            'position' => 1415
        ),
        array(
            'group' => 'images',
            'name'  => 'sCatIconsize',
            'type'  => 'str',
            'value' => '168*100',
            'position' => 1420
        ),
        array(
            'group' => 'images',
            'name'  => 'sCatPromotionsize',
            'type'  => 'str',
            'value' => '370*107',
            'position' => 1425
        ),
        array(
            'group' => 'images',
            'name'  => 'sCatThumbnailsize',
            'type'  => 'str',
            'value' => '1140*250',
            'position' => 1430
        ),
        array(
            'group' => 'images',
            'name'  => 'sManufacturerIconsize',
            'type'  => 'str',
            'value' => '100*100',
            'position' => 1435
        ),
        array(
            'group' => 'images',
            'name'  => 'blSliderShowImageCaption',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1440
        ),

        //Group background:
        array(
            'group' => 'background',
            'name'  => 'sBackgroundColor',
            'type'  => 'str',
            'value' => '#CCEBF5',
            'position' => 1500
        ),
        array(
            'group' => 'background',
            'name'  => 'blUseBackground',
            'type'  => 'bool',
            'value' => 0,
            'position' => 1505
        ),
        array(
            'group' => 'background',
            'name'  => 'sBackgroundPath',
            'type'  => 'str',
            'value' => 'mein_hintergrundbild.jpg',
            'position' => 1510
        ),
        array(
            'group'       => 'background',
            'name'        => 'sBackgroundSize',
            'type'        => 'select',
            'value'       => 'cover',
            'constraints' => 'cover|contain|normal',
            'position' => 1515
        ),
        array(
            'group'       => 'background',
            'name'        => 'sBackgroundPosHorizontal',
            'type'        => 'select',
            'value'       => 'center',
            'constraints' => 'left|right|center',
            'position' => 1520
        ),
        array(
            'group'       => 'background',
            'name'        => 'sBackgroundPosVertical',
            'type'        => 'select',
            'value'       => 'top',
            'constraints' => 'top|bottom|center',
            'position' => 1525
        ),
        array(
            'group'       => 'background',
            'name'        => 'sBackgroundRepeat',
            'type'        => 'select',
            'value'       => 'no-repeat',
            'constraints' => 'no-repeat|repeat-x|repeat-y|repeat',
            'position' => 1530
        ),
        array(
            'group' => 'background',
            'name'  => 'blBackgroundAttachment',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1535
        ),

        //Group features:
        array(
            'group' => 'features',
            'name'  => 'bl_showCompareList',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1600
        ),
        array(
            'group' => 'features',
            'name'  => 'bl_showPriceAlarm',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1605
        ),
        array(
            'group' => 'features',
            'name'  => 'bl_showWishlist',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1610
        ),
        array(
            'group' => 'features',
            'name'  => 'bl_showListmania',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1615
        ),
        array(
            'group' => 'features',
            'name'  => 'bl_showVouchers',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1620
        ),
        array(
            'group' => 'features',
            'name'  => 'bl_showGiftWrapping',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1625
        ),

        //Group footer:
        array(
            'group' => 'footer',
            'name'  => 'blFooterShowHelp',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1700
        ),
        array(
            'group' => 'footer',
            'name'  => 'blFooterShowLinks',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1705
        ),
        array(
            'group' => 'footer',
            'name'  => 'blFooterShowNewsletter',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1710
        ),
        array(
            'group' => 'footer',
            'name'  => 'blFooterShowNewsletterForm',
            'type'  => 'bool',
            'value' => 1,
            'position' => 1715
        ),
        array(
            'group' => 'footer',
            'name'  => 'sFacebookUrl',
            'type'  => 'str',
            'value' => 'https://www.facebook.com',
            'position' => 1720
        ),
        array(
            'group' => 'footer',
            'name'  => 'sTwitterUrl',
            'type'  => 'str',
            'value' => 'https://twitter.com',
            'position' => 1725
        ),
        array(
            'group' => 'footer',
            'name'  => 'sYouTubeUrl',
            'type'  => 'str',
            'value' => 'https://www.youtube.com',
            'position' => 1730
        ),
        array(
            'group' => 'footer',
            'name'  => 'sBlogUrl',
            'type'  => 'str',
            'value' => 'https://wordpress.org',
            'position' => 1735
        ),

        //Group contact:
        array(
            'group' => 'contact',
            'name'  => 'sGoogleMapsAddr',
            'type'  => 'str',
            'value' => 'OXID eSales AG, Bertoldstraße 48, 79098 Freiburg',
            'position' => 1800
        ),

        //Group emails:
        array(
            'group' => 'emails',
            'name'  => 'sEmailLogo',
            'type'  => 'str',
            'value' => 'logo_email.png',
            'position' => 1900
        ),
        array(
            'group' => 'emails',
            'name'  => 'blEmailsShowProductPictures',
            'type'  => 'bool',
            'value' => 0,
            'position' => 1905
        ),

        //Group googleanalytics:
        array(
            'group' => 'googleanalytics',
            'name'  => 'sGATrackingId',
            'type'  => 'str',
            'value' => '',
            'position' => 2000
        ),
        array(
            'group' => 'googleanalytics',
            'name'  => 'blUseGAEcommerceTracking',
            'type'  => 'bool',
            'value' => 1,
            'position' => 2005
        ),
        array(
            'group' => 'googleanalytics',
            'name'  => 'blUseGAPageTracker',
            'type'  => 'bool',
            'value' => 1,
            'position' => 2010
        ),
        array(
            'group' => 'googleanalytics',
            'name'  => 'blGAAnonymizeIPs',
            'type'  => 'bool',
            'value' => 1,
            'position' => 2015
        ),

        //Group googlets:
        array(
            'group' => 'googlets',
            'name'  => 'blUseGoogleTS',
            'type'  => 'bool',
            'value' => 0,
            'position' => 2100
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sGoogleVendorId',
            'type'  => 'str',
            'value' => '',
            'position' => 2105
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sPageLanguage',
            'type'  => 'str',
            'value' => 'de_DE',
            'position' => 2110
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sGoogleShoppingAccountId',
            'type'  => 'str',
            'value' => '',
            'position' => 2115
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sShippingDaysOnStock',
            'type'  => 'str',
            'value' => '3',
            'position' => 2120
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sShippingDaysNotOnStock',
            'type'  => 'str',
            'value' => '5',
            'position' => 2125
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sDeliveryDaysOnStock',
            'type'  => 'str',
            'value' => '6',
            'position' => 2130
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sDeliveryDaysNotOnStock',
            'type'  => 'str',
            'value' => '14',
            'position' => 2135
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sShoppingLanguage',
            'type'  => 'str',
            'value' => 'de',
            'position' => 2140
        ),
        array(
            'group' => 'googlets',
            'name'  => 'sShoppingCountry',
            'type'  => 'str',
            'value' => 'DE',
            'position' => 2145
        ),
    ),
);
